src/classmentors/components/cohorts/cohorts.js

```diff
@@ -1036,31 +925,113 @@ function ClmCohortRankPageCtrl($q, $scope, $log, firebaseApp, $firebaseObject, $
                         oneEventData.title = result.title;
                         self.cohortTotalParticipants += oneEventData.participants.length;
                         oneEventData.id = result.$id;
-                        self.cohortEventData.push(oneEventData);
-                        iter++;
-                        loopDBEvents();
+                        oneEventData.userRanks = [];
+                        var userIndex = 0;
+                        loadUserAchievements();
+                        function loadUserAchievements() {
+                            if(userIndex < oneEventData.participants.length) {
+                                var participantRankingRef = db.ref(`classMentors/userProfiles/${oneEventData.participants[userIndex].$id}`);
+                                var participantRankingObj = $firebaseObject(participantRankingRef);
+                                participantRankingObj.$loaded().then(function () {
+                                    var rankingResult = participantRankingObj;
+                                    console.log(rankingResult);
+                                    if(rankingResult.services && rankingResult.services.freeCodeCamp) {
+                                        oneEventData.userRanks.push({"user": rankingResult.user, "total": parseInt(rankingResult.services.freeCodeCamp.totalAchievements)});
+                                    } else {
+                                        oneEventData.userRanks.push({"user": rankingResult.user, "total": 0});
+                                    }
+                                    userIndex++;
+                                    loadUserAchievements();
```

Comment:
It's not obviously what the function is meant to do. You might only notice at the end it's an iteration and personalty I didn't notice it was meant to be sequential. A `for` loop, `forEach` or `reduce` AFAIK is more idiomatic in this case.

Is there a noticeable performance issue when doing it in parallel?